controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -669,3 +724,160 @@ bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessag
+{
+    static const int max_tlv                          = 256; // Magic number
+    size_t tlv_idx                                    = 0;
+    bool skip                                         = false;
+    bool finish                                       = false;
+    static const std::vector<std::string> tlv_members = {"tlv_type", "tlv_length", "tlv_value"};
+
+    do {
+        tlv_hex_t tlv_hex;
+        for (size_t tlv_member_idx = 0; tlv_member_idx < tlv_members.size(); tlv_member_idx++) {
+            std::string lookup_str =
+                tlv_members[tlv_member_idx] + (tlv_idx > 0 ? std::to_string(tlv_idx) : "");
+            auto it = params.find(lookup_str);
+            if (it == params.end()) {
+                if (tlv_idx == 0 && tlv_member_idx == 0) {
+                    // Skipping TLV with no index (index 0), checking TLV index 1
+                    skip = true;
+                    break;
+                } else if (tlv_idx > 0 && tlv_member_idx == 0) {
+                    // No indexed TLVs found, finish
+                    finish = true;
+                    break;
+                } else {
+                    err_string = "missing param name '" + lookup_str + "'";
+                    return false;
+                }
+            }
+
+            // Type
+            if (tlv_member_idx == 0) {
+                tlv_hex.type = &it->second;
+
+                // Length
+            } else if (tlv_member_idx == 1) {
+                tlv_hex.length = &it->second;
+
+                // Value
+            } else if (tlv_member_idx == 2) {
+                // The hex values contains '{' and '}' delimiters, but we don't care about them.
+                it->second.erase(std::remove(it->second.begin(), it->second.end(), '{'),
+                                 it->second.end());
+                it->second.erase(std::remove(it->second.begin(), it->second.end(), '}'),
+                                 it->second.end());
+                tlv_hex.value = &it->second;
+            } else {
+                LOG(ERROR) << "Illegal tlv_member_idx value: " << int(tlv_member_idx);
+                err_string = err_internal;
+                return false;
+            }
+
+            // Validate hex notation on value or list of values separated by space
+            auto values = string_utils::str_split(it->second, ' ');
+            for (auto value : values) {
+                if (!validate_hex_notation(value)) {
+                    err_string = "param name '" + lookup_str + "' has invalid hex notation value";
+                    return false;
+                }
+
+                // Type and Length fields must have fixed size:
+
+                // Type
+                if (tlv_member_idx == 0 && value.size() != 4) {
+                    err_string = "type field is too large: " + value;
+                    return false;
+
+                    // Length
+                } else if (tlv_member_idx == 1 && value.size() != 6) {
```

Comment:
I'm not sure if this will always be 4 digits. It's possible that UCC will just print the significant digits. Although the examples indeed always have 4 digits.

Reply:
I think it must have 4 digits because how the data is presented in the command is the way we should write it to the buffer. Again if I'm wrong, I will fix in the future.